### PR TITLE
OHRM5X-1479: Optimize Attachment Fetching With Entity Framework

### PR DESCRIPTION
**Pull request by SandeepaOHRM**

### CODE REVIEW

File: src/plugins/orangehrmRecruitmentPlugin/Api/AbstractCandidateActionAPI.php [hunk 1]
```diff
@@ -232,7 +232,9 @@ protected function getInterviewId(): ?int
     protected function setCandidateAsEmployee(CandidateVacancy $candidateVacancy, Employee $employee)
     {
         $employee->setFirstName($candidateVacancy->getCandidate()->getFirstName());
-        $employee->setMiddleName(!is_null($candidateVacancy->getCandidate()->getMiddleName()) ?: '');
+        $middleName = !is_null($candidateVacancy->getCandidate()->getMiddleName()) ?
+            $candidateVacancy->getCandidate()->getMiddleName() : '';
+        $employee->setMiddleName($middleName);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Api/AbstractCandidateActionAPI.php:237]:
```suggestion
        $employee->setMiddleName($candidateVacancy->getCandidate()->getMiddleName() ?? '');
```

File: src/plugins/orangehrmRecruitmentPlugin/Api/Model/VacancyAttachmentModel.php [hunk 2]
```diff
@@ -21,24 +21,23 @@
...
 
-    public function __construct(VacancyAttachment $vacancyAttachment)
+    public function __construct(RecruitmentAttachment $vacancyAttachment)
     {
         $this->setEntity($vacancyAttachment);
         $this->setFilters(
             [
                 'id',
-                ['getVacancy','getId'],
+                'fkIdentity',
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Api/Model/VacancyAttachmentModel.php:36]:
```suggestion
                'fkIdentity', // vacancy id
```

File: src/plugins/orangehrmRecruitmentPlugin/Api/Model/VacancyAttachmentModel.php [hunk 3]
```diff
@@ -48,8 +47,7 @@ public function __construct(VacancyAttachment $vacancyAttachment)
                 ['attachment', 'fileName'],
                 ['attachment', 'fileType'],
                 ['attachment', 'fileSize'],
-                'comment',
-                'attachmentType',
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Api/Model/VacancyAttachmentModel.php:52]: any idea why `attachmentType` removed?
if this is required attribute you may have to create child class of `RecruitmentAttachment` and add `attachmentType` attribute

File: src/plugins/orangehrmRecruitmentPlugin/Dao/RecruitmentAttachmentDao.php [hunk 4]
```diff
@@ -73,11 +75,25 @@ public function getVacancyAttachmentById(int $attachId): ?VacancyAttachment
         return null;
     }
 
+    /**
+     * @param int $candidateId
+     * @return RecruitmentAttachment|null
+     */
+    public function getCandidateAttachmentByCandidateId(int $candidateId): ?RecruitmentAttachment
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Dao/RecruitmentAttachmentDao.php:82]:
```suggestion
    public function getPartialCandidateAttachmentByCandidateId(int $candidateId): ?RecruitmentAttachment
```
https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/partial-objects.html
This is a deprecated pattern. but we are going to achieve this

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmRecruitmentPlugin/Api/AbstractCandidateActionAPI.php (hunk 1 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Api/AbstractCandidateActionAPI.php:237]: fixed

File: src/plugins/orangehrmRecruitmentPlugin/Api/Model/CandidateAttachmentModel.php [hunk 5]
```diff
@@ -21,19 +21,19 @@
...
 
-    public function __construct(CandidateAttachment $candidateAttachment)
+    public function __construct(RecruitmentAttachment $candidateAttachment)
     {
         $this->setEntity($candidateAttachment);
         $this->setFilters(
             [
                 'id',
-                ['getCandidate', 'getId'],
+                'fkIdentity',
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Api/Model/CandidateAttachmentModel.php:36]: done

File: src/plugins/orangehrmRecruitmentPlugin/Api/Model/InterviewAttachmentModel.php [hunk 6]
```diff
@@ -21,19 +21,19 @@
...
 
-    public function __construct(InterviewAttachment $interviewAttachment)
+    public function __construct(RecruitmentAttachment $interviewAttachment)
     {
         $this->setEntity($interviewAttachment);
         $this->setFilters(
             [
                 'id',
-                ['getInterview', 'getId'],
+                'fkIdentity',
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Api/Model/InterviewAttachmentModel.php:36]: done

File: src/plugins/orangehrmRecruitmentPlugin/Api/Model/VacancyAttachmentModel.php (hunk 2 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Api/Model/VacancyAttachmentModel.php:36]: done

File: src/plugins/orangehrmRecruitmentPlugin/Api/Model/VacancyAttachmentModel.php (hunk 3 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Api/Model/VacancyAttachmentModel.php:52]: this model was the first one I created in my internship time. Checked through the code base and there is no use of attachmentType attribute and removed it.

File: src/plugins/orangehrmRecruitmentPlugin/Controller/File/VacancyAttachment.php [hunk 7]
```diff
@@ -34,7 +34,7 @@ public function handle(Request $request): Response
         $response = $this->getResponse();
 
         if ($attachId) {
-            $attachment = $this->getRecruitmentAttachmentService()->getRecruitmentAttachmentDao()->getVacancyAttachmentById(
+            $attachment = $this->getRecruitmentAttachmentService()->getRecruitmentAttachmentDao()->getVacancyAttachmentContentById(
                 $attachId
             );
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Controller/File/VacancyAttachment.php:39]: oops didn't see this

File: src/plugins/orangehrmAdminPlugin/Dto/JobSpecificationAttachmentDetails.php [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Admin\Dto;
+
+class JobSpecificationAttachmentDetails
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmAdminPlugin/Dto/JobSpecificationAttachmentDetails.php:22]: done

File: src/plugins/orangehrmRecruitmentPlugin/Dao/RecruitmentAttachmentDao.php [hunk 9]
```diff
@@ -73,11 +75,25 @@ public function getVacancyAttachmentById(int $attachId): ?VacancyAttachment
...
+        $qb->setParameter('candidateId', $candidateId);
+        return $qb->getQuery()->getOneOrNullResult();
+    }
+
     /**
      * @param int $candidateId
      * @return CandidateAttachment|null
      */
-    public function getCandidateAttachmentByCandidateId(int $candidateId): ?CandidateAttachment
+    public function getCandidateAttachmentContentByCandidateId(int $candidateId): ?CandidateAttachment
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Dao/RecruitmentAttachmentDao.php:96]: done

File: src/plugins/orangehrmRecruitmentPlugin/Dao/RecruitmentAttachmentDao.php (hunk 4 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Dao/RecruitmentAttachmentDao.php:82]: done

**Comment by SandeepaOHRM** [comment]
Closed since the origin branch was created tracking wrong upstream branch.

**Comment by RajithaKumara** [comment]
Moved to #1484

**Pull request closed without merging**